frameworks/elastic/docs/install.md

```diff
@@ -79,3 +79,49 @@ Elastic supports deployment on virtual networks on DC/OS (including the `dcos` o
 }
 ```
 As mentioned in the [developer guide](https://mesosphere.github.io/dcos-commons/developer-guide.html) once the service is deployed on a virtual network, it cannot be updated to use the host network.
+
+## TLS
+
+The Elastic service can be launched with TLS encryption. Enabling TLS will switch all internal communication between Elastic nodes to encrypted connections.
+
+Enabling TLS is possible only in `permissive` and `strict` cluster security modes. Both modes require a service account. Additionally, a service account must have the `dcos:superuser` permission. If the permission is missing the Elastic scheduler will not abe able to provision TLS artifacts.
+
+Installing Elastic with TLS support requires enabling the [X-Pack](x-pack.md).
+
+Sample JSON options file named `elastic-tls.json`:
+```json
+{
+  "service": {
+    "service_account_secret": "elastic",
+    "service_account": "elastic",
+    "tls": true
+  },
+  "elasticsearch": {
+    "xpack_enabled": true
+  }
+}
+```
+
+For more information about TLS in the SDK see [the TLS documentation](https://mesosphere.github.io/dcos-commons/developer-guide.html#tls).
+
+### Clients
+
+Clients connecting to the Elastic service are required to use [the DC/OS CA bundle](https://docs.mesosphere.com/1.10/networking/tls-ssl/get-cert/) to verify the TLS connections.
+
+### Kibana
+
+When the Elastic service is deployed on DC/OS with TLS support the Kibana, acting as a Elastic client, must be configured to verify TLS with the DC/OS CA bundle. To install the DC/OS CA bundle launch the Kibana with following configuration.
+
+Sample JSON options file named `kibana-tls.json`:
+```json
+{
+    "kibana": {
+        "xpack_enabled": true,
+        "elasticsearch_url": "https://coordinator.elastic.l4lb.thisdcos.directory:9200",
+        "elasticsearch_tls": true
+        ...
```

Comment:
We've done this elsewhere to get markdown to stop complaining about valid json formatting:

```json
{
    "kibana": {
        "xpack_enabled": true,
        "elasticsearch_url": "https://coordinator.elastic.l4lb.thisdcos.directory:9200",
        "elasticsearch_tls": true,
        "...": "..."
    }
}
```